main.go

```diff
@@ -26,84 +24,74 @@ var (
+const serviceName = "prometheus-service"
+const envVarLogLevel = "LOG_LEVEL"
+const monitoringTriggeredEvent = "sh.keptn.event.configure-monitoring.triggered"
+const getSliTriggeredEvent = "sh.keptn.event.get-sli.triggered"
 
-	// start internal CloudEvents handler (on port env.Port)
-	os.Exit(_main(env))
-}
-
-func _main(env utils.EnvConfig) int {
-	ctx := context.Background()
-	ctx = cloudevents.WithEncodingStructured(ctx)
-
-	p, err := cloudevents.NewHTTP()
-	if err != nil {
-		log.Fatalf("failed to create protocol: %s", err.Error())
+func main() {
+	if os.Getenv(envVarLogLevel) != "" {
+		logLevel, err := logrus.ParseLevel(os.Getenv(envVarLogLevel))
+		if err != nil {
+			logrus.WithError(err).Error("could not parse log level provided by 'LOG_LEVEL' env var")
+			logrus.SetLevel(logrus.InfoLevel)
+		} else {
+			logrus.SetLevel(logLevel)
+		}
 	}
 
-	ceHandler, err := cloudevents.NewHTTPReceiveHandler(ctx, p, gotEvent)
-	if err != nil {
-		log.Fatalf("failed to create handler: %s", err.Error())
-	}
+	log.Printf("Starting %s", serviceName)
 
 	http.HandleFunc("/", HTTPGetHandler)
-	http.Handle(env.Path, ceHandler)
-	err = http.ListenAndServe(":8080", nil)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	return 0
-}
+	go func() {
+		err := http.ListenAndServe(":8080", nil)
+		if err != nil {
+			log.Fatalf("Error with HTTP server: %e", err)
+		}
+	}()
 
-// gotEvent processes an incoming CloudEvent
-func gotEvent(event cloudevents.Event) error {
-	keptnContext, err := event.Context.GetExtension("shkeptncontext")
+	clusterConfig, err := rest.InClusterConfig()
 	if err != nil {
-		return fmt.Errorf("cloud event does not contain the field 'shkeptncontext'")
+		log.Fatalf("unable to create kubernetes cluster config: %e", err)
 	}
 
-	shkeptncontext, err := types.ToString(keptnContext)
+	kubeClient, err := kubernetes.NewForConfig(clusterConfig)
 	if err != nil {
-		return fmt.Errorf("field 'shkeptncontext' can not be parsed as keptn context")
-	}
-
-	logger := keptncommon.NewLogger(shkeptncontext, "", utils.ServiceName)
-
-	// convert v0.1.4 spec monitoring.configure CloudEvent into a v0.2.0 spec configure-monitoring.triggered CloudEvent
-	if event.Type() == keptn.ConfigureMonitoringEventType {
-		event.SetType(keptnv2.GetTriggeredEventType(keptnv2.ConfigureMonitoringTaskName))
+		log.Fatalf("unable to create kubernetes client: %e", err)
 	}
 
-	keptnHandler, err := keptnv2.NewKeptn(&event, keptncommon.KeptnOpts{})
+	log.Fatal(sdk.NewKeptn(
+		serviceName,
+		sdk.WithTaskHandler(
+			monitoringTriggeredEvent,
+			eventhandling.NewConfigureMonitoringEventHandler(),
+			prometheusTypeFilter),
+		sdk.WithTaskHandler(
+			getSliTriggeredEvent,
+			eventhandling.NewGetSliEventHandler(*kubeClient),
+			prometheusSLIProviderFilter),
+		sdk.WithLogger(logrus.New()),
+	).Start())
+}
 
-	if err != nil {
-		return fmt.Errorf("could not create Keptn handler: %v", err)
+func prometheusSLIProviderFilter(keptnHandle sdk.IKeptn, event sdk.KeptnEvent) bool {
```

Comment:
same here, please add a comment like "filtering get-sli.triggered events for prometheus"